docs/dev/guidelines/database.rst

```diff
@@ -0,0 +1,226 @@
+
+ .. code-block:: java
+
+    // AnswerCounter.java
+    @OneToOne(cascade = { CascadeType.PERSIST })
+    @JoinColumn(unique = true)
+    private AnswerOption answer;
+
+
+* ``CascadeType.MERGE`` If you merge the source entity (saved/updated/synchronized) to the database, the merge is cascaded to the target of the association. This rule applies to existing objects only. Use this type to always merge/synchronize the existing data in the table with the data in the object. Example below: whenever we merge a ``Result`` to the database, i.e. save the changes on the object, the ``Assessor`` object is also merged/saved. 
+
+ .. code-block:: java
+
+    // Result.java
+    @OneToOne(cascade = CascadeType.MERGE, fetch = FetchType.LAZY)
+    @JoinColumn(unique = false)
+    private User assessor;
+
+
+* ``CascadeType.REMOVE`` If the source entity is removed, the target of the association is also removed. Example below: propagates **remove** operation from parent ``Submission`` to child ``Result``. When a ``Submission`` is deleted, the corresponding ``Result`` is also deleted.
+
+ .. code-block:: java
+
+    // Submission.java
+    @OneToOne(mappedBy = "submission", fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
+    @JsonIgnoreProperties({ "submission", "participation" })
+    @JoinColumn(unique = true)
+    private Result result;
+ 
+
+* ``CascadeType.REFRESH`` If the source entity is refreshed, it cascades the refresh to the target of the association. This is used to refresh the data in the object and its associations. This is useful for cases where there is a change which needs to be synchronized FROM the database.
+
+Not used in Artemis yet.
+
+
+Best Practices
+==============
+* If you want to create a ``@OneToMany`` relationship or ``@ManyToMany`` relationship, first think about if it is important for the association to be ordered. If you do not need the association to be ordered, then always go for a ``Set`` instead of ``List``. If you are unsure, start with a ``Set``. 
+
+  * **Unordered Collection**: A ``Set`` comes with certain advantages such as ensuring that there are no duplicates and null values in your collection. There are also performance arguments to use a ``Set``, especially for ``@ManyToMany`` relationships. For more information see this `stackoverflow thread <https://stackoverflow.com/questions/4655392/which-java-type-do-you-use-for-jpa-collections-and-why>`_. E.g.:
+
+       .. code-block:: java
+
+        // Course.java
+        @OneToMany(mappedBy = "course", fetch = FetchType.LAZY)
+        @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+        @JsonIgnoreProperties("course")
+        private Set<Exercise> exercises = new HashSet<>();
+
+
+.. _ordered:
```

Comment:
Why is this `.. ordered:` needed?

Reply:
This is a way to label specific sections in Markdown. We use this to reference this section from within the text.